### PR TITLE
Ensure email address is set when using SSO

### PR DESCRIPTION
**Pull request by Ntu2koTrevor**

### CODE REVIEW

**Comment by Ntu2koTrevor** [comment]
@miltontony Please review

File: molo/core/backends.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ def authenticate(self, ticket, service, request):
             and 'has_perm' in request.session['attributes']\
                 and request.session['attributes']['has_perm'] == 'True':
             if request.session['attributes']['is_admin'] == 'True':
+                user.email
```
Comment (miltontony) [inline molo/core/backends.py]: you need to get this value from the session. And you'll need a test as well.

**Comment by Ntu2koTrevor** [comment]
@miltontony please re-review

File: molo/core/backends.py [hunk 2]
```diff
@@ -13,7 +13,8 @@ def authenticate(self, ticket, service, request):
         if 'attributes' in request.session \
             and 'has_perm' in request.session['attributes']\
                 and request.session['attributes']['has_perm'] == 'True':
-            if request.session['attributes']['is_admin'] == 'True':
+            if request.session['attributes']['is_admin'] == 'True' and \
+               request.session['attributes']['email'] == user.email:
```
Comment (miltontony) [inline molo/core/backends.py]: This is not right, it should not be in the if statement.
You're looking for
```python
user.email = request.session['attributes']['email'] = user.email
```

**Comment by Ntu2koTrevor** [comment]
@miltontony ^^^

File: molo/core/tests/test_cas.py [hunk 3]
```diff
@@ -84,7 +84,7 @@ def test_succesful_login(self, mock_verify):
         mock_verify.return_value = (
             'test@example.com',
             {'ticket': 'fake-ticket', 'service': service, 'has_perm': 'True',
-             'is_admin': 'True'},
+             'is_admin': 'True', 'email': 'test@example.com'},
```
Comment (miltontony) [inline molo/core/tests/test_cas.py]: This works, but I'd also like to get another test that checks that the user that's created has the email field set.

Comment (miltontony) [inline molo/core/tests/test_cas.py]: So on `line 94`, can you get the user `User.objects.get(username='test@example.com')` and assert that the returned user has the email address on the email field.

**Comment by Ntu2koTrevor** [comment]
@miltontony ^^^

**Comment by miltontony** [comment]
👍 🍰 cc: @saeedmb8

**Pull request merged**